packages/components/atoms/f-segmented-control/src/components/SegmentedControl.vue

```diff
@@ -0,0 +1,254 @@
+            // so users can easily return to their last focused item when tabbing back.
+            if (event.key === 'ArrowRight' || event.key === 'ArrowDown') {
+                event.preventDefault();
+                // Move focus to the next button
+                newIndex = (this.tabIndex + 1) % this.options.length;
+            } else if (event.key === 'ArrowLeft' || event.key === 'ArrowUp') {
+                event.preventDefault();
+                // Move focus to the previous button
+                newIndex = (this.tabIndex - 1 + this.options.length) % this.options.length;
+            } else if (event.key === 'Enter' || event.key === 'Space') {
+                event.preventDefault();
+                this.selectOption(this.options[this.tabIndex].label);
+
+                return;
+            }
+
+            // If the newIndex has changed, update the tabIndex and focus the new button
+            if (newIndex !== this.tabIndex) {
+                this.focusNewButton(newIndex, currentButton);
+            }
+        },
+        focusNewButton (newIndex, currentButton) {
+            this.tabIndex = newIndex;
+            // Remove focus from the current button
+            currentButton.blur();
+
+            // Wait for the DOM to update, then focus the new button
+            this.$nextTick(() => {
+                this.$el.children[this.tabIndex].focus();
+            });
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+@use '@justeat/fozzie/src/scss/fozzie' as f;
+
+// Variables
+$sc-sm-height: f.spacing(f);
+$sc-l-height: calc(f.spacing(b) * 6);
+$sc-border-radius: 9999px;
+$sc-padding-xs-modifier: calc(f.spacing(a) / 2);
+$sc-background-clr: f.$color-container-strong;
+$sc-background-clr-selected: f.$color-container-default;
+
+// Segmented control container
+.c-segmented-control {
+    display: flex;
+    justify-content: space-between;
+    background-color: $sc-background-clr;
+    border-radius: $sc-border-radius;
+    gap: 2.5px; // Ensure outlines and box-shadows don't overlap
+    padding: $sc-padding-xs-modifier;
+    min-block-size: $sc-sm-height; // Small size by default
+    user-select: none;
+
+    // Large size modifier
+    &--large {
+        min-block-size: $sc-l-height;
+
+        .c-segmented-control__option {
+            padding-block: calc($sc-padding-xs-modifier + f.spacing(b));
+        }
+    }
+}
+
+// Segmented control options
+.c-segmented-control__option {
+    font-size: inherit;
+    color: inherit;
+    flex: 1; // Distribute available space equally among buttons
+    border-radius: inherit;
+    border: none;
+    padding: f.spacing(a) f.spacing(e);
+    font-weight: f.$font-weight-regular;
+    background: none;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    gap: f.spacing(b);
+    outline: none; // Replace outline with box-shadow on focus due to iOS Safari not respecting border-radius
+
+    // Selected state
+    &--selected {
```

Comment:
Could you use the full selector here please as it makes it easier to find and change selectors in future.

Reply:
I'm happy to do this but I personally disagree with the value it adds 😂

Reply:
I do prefer the way you did it originally but it adds ambiguity code base imo. Otherwise i have no issues with it, i think this came up a few times over the years so we've always stuck to the full name for consistency.

Reply:
I like the syntax too – but it does make searchability harder when looking up class names in the codebase, which is why we don't use them